project/Boilerplate.scala

```diff
@@ -0,0 +1,224 @@
+    override def range: IndexedSeq[Int] = 1 to maxArity
+
+    def filename(root: File): File = root / "algebra" / "std" / "TupleAlgebra.scala"
+
+    def content(tv: TemplateVals): String = {
+      import tv._
+
+      def constraints(constraint: String) =
+        synTypes.map(tpe => s"${tpe}: ${constraint}[${tpe}]").mkString(", ")
+
+      def tuple(results: TraversableOnce[String]) = {
+        val resultsVec = results.toVector
+        val a = synTypes.size
+        val r =  s"${0.until(a).map(i => resultsVec(i)).mkString(", ")}"
+        if (a == 1) "Tuple1(" ++ r ++ ")"
+        else s"(${r})"
+      }
+
+      def binMethod(name: String) =
+        synTypes.zipWithIndex.iterator.map {
+          case (tpe, i) =>
+            val j = i + 1
+            s"${tpe}.${name}(x._${j}, y._${j})"
+        }
+
+      def binTuple(name: String) =
+        tuple(binMethod(name))
+
+      def unaryTuple(name: String) = {
+        val m = synTypes.zipWithIndex.map { case (tpe, i) => s"${tpe}.${name}(x._${i + 1})" }
+        tuple(m)
+      }
+
+      def nullaryTuple(name: String) = {
+        val m = synTypes.map(tpe => s"${tpe}.${name}")
+        tuple(m)
+      }
+
+      block"""
+        |package algebra
+        |package std
+        |
+        |import algebra.ring.{EuclideanRing, Rig, Ring, Rng, Semiring}
+        |
+        |private[std] trait TupleInstances {
+        -  implicit def tuple${arity}Group[${`A..N`}](implicit ${constraints("Group")}): Group[${`(A..N)`}] =
+        -    new Group[${`(A..N)`}] {
+        -      def combine(x: ${`(A..N)`}, y: ${`(A..N)`}): ${`(A..N)`} = ${binTuple("combine")}
+        -      def empty: ${`(A..N)`} = ${nullaryTuple("empty")}
+        -      def inverse(x: ${`(A..N)`}): ${`(A..N)`} = ${unaryTuple("inverse")}
+        -    }
+        -
+        -  implicit def tuple${arity}Eq[${`A..N`}](implicit ${constraints("Eq")}): Eq[${`(A..N)`}] =
+        -    new Eq[${`(A..N)`}] {
+        -      def eqv(x: ${`(A..N)`}, y: ${`(A..N)`}): Boolean = ${binMethod("eqv").mkString(" && ")}
+        -    }
+        -
+        -  implicit def tuple${arity}EuclideanRing[${`A..N`}](implicit ${constraints("EuclideanRing")}): EuclideanRing[${`(A..N)`}] =
+        -    new EuclideanRing[${`(A..N)`}] {
+        -      def mod(x: ${`(A..N)`}, y: ${`(A..N)`}): ${`(A..N)`} = ${binTuple("mod")}
+        -      def negate(x: ${`(A..N)`}): ${`(A..N)`} = ${unaryTuple("negate")}
+        -      def one: ${`(A..N)`} = ${nullaryTuple("one")}
+        -      def plus(x: ${`(A..N)`}, y: ${`(A..N)`}): ${`(A..N)`} = ${binTuple("plus")}
+        -      def quot(x: ${`(A..N)`}, y: ${`(A..N)`}): ${`(A..N)`} = ${binTuple("quot")}
+        -      def times(x: ${`(A..N)`}, y: ${`(A..N)`}): ${`(A..N)`} = ${binTuple("times")}
+        -      def zero: ${`(A..N)`} = ${nullaryTuple("zero")}
+        -    }
+        -
+        -  implicit def tuple${arity}Field[${`A..N`}](implicit ${constraints("Field")}): Field[${`(A..N)`}] =
```

Comment:
I think I suggested that @adelbertc add this, but yes, we should remove it. Sorry!